controllers/clustergroupupgrade_controller.go

```diff
@@ -281,24 +266,189 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 	return nextReconcile, nil
 }
 
+func (r *ClusterGroupUpgradeReconciler) initializeRemediationPolicyForBatch(
+	clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade) {
+
+	if clusterGroupUpgrade.Status.Status.CurrentBatchStartedAt.IsZero() {
+		clusterGroupUpgrade.Status.Status.RemediationPlanForBatch = make(map[string]int)
+		batchIndex := clusterGroupUpgrade.Status.Status.CurrentBatch - 1
+
+		// By default, assume all clusters start applying all policies from the first policy.
+		for _, batchClusterName := range clusterGroupUpgrade.Status.RemediationPlan[batchIndex] {
+			clusterGroupUpgrade.Status.Status.RemediationPlanForBatch[batchClusterName] = 0
+		}
+
+		// Set the time for when the batch started updating.
+		clusterGroupUpgrade.Status.Status.CurrentBatchStartedAt = metav1.Now()
+
+		r.Log.Info(">>>> InitializeRemediationPolicyForBatch ", "remediationPlanForBatch", clusterGroupUpgrade.Status.Status.RemediationPlanForBatch)
+		r.Log.Info("NEW CurrentBatchStartedAt", "clusterGroupUpgrade.Status.Status.CurrentBatchStartedAt", clusterGroupUpgrade.Status.Status.CurrentBatchStartedAt)
+	}
+}
+
+func (r *ClusterGroupUpgradeReconciler) getNextRemediationPoliciesForBatch(
```

Comment:
Yes, I like that. Will add. I have that for another function below. It would be nice to have something like this for every function. Will add it as part of the cleanup.

Reply:
Done